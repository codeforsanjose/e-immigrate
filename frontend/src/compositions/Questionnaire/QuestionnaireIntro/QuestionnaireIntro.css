.QuestionnaireIntro {
    margin: 0 10%;
}

.QuestionnaireIntro h1 {
    font-size: 1.5rem;
    font-weight: 700;
    margin: 0 0 1rem 0;
}

.QuestionnaireIntro h2 {
    font-size: 17px;
    font-weight: 400;
    margin: 0 0 0.5rem 0;
}

.QuestionnaireIntro ul {
    list-style-image: url('../../../data/images/CIIT_bullet.svg');
    margin-left: 16px;
}

.QuestionnaireIntro .ButtonContainer {
    margin: 48px 0 16px;
    display: grid;
}

.QuestionnaireIntro .header2 {
    margin-top: 40px;
}
