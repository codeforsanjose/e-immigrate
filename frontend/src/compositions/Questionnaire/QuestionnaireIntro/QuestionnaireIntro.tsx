import React from 'react';
import { Button } from '../../../components/Button/Button';

import './QuestionnaireIntro.css';
import { useContentContext } from '../../../contexts/ContentContext';

type QuestionnaireIntroProps = {
    setIntroPage: (value: boolean) => void;
};
export function QuestionnaireIntro(props: QuestionnaireIntroProps) {
    const {
        setIntroPage,
    } = props;
    const { content } = useContentContext();
    const onClick = () => {
        setIntroPage(false);
    };

    return (
        <div className="QuestionnaireIntro">
            <div>
                <h1>
                    {content.step2Header}: {content.step2Title}
                </h1>
                <h2>{content.step2Instructions}</h2>
                <ul>
                    <li>{content.step2Tip1}</li>
                    <li>{content.step2Tip2}</li>
                </ul>
            </div>
            <div>
                <h1 className="header2">{content.step2Header2}</h1>
                <ul>
                    <li>{content.step2Tip4}</li>
                    <li>{content.step2Tip5}</li>
                    <li>{content.step2Tip6}</li>
                </ul>
            </div>
            <div className="ButtonContainer">
                <Button label={content.step2ProceedButton1} onClick={onClick} />
            </div>
        </div>
    );
}
