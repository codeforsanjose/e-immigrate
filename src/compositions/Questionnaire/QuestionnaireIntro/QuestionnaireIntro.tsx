import React from 'react';
import { Button } from '../../../components/Button/Button';

import './QuestionnaireIntro.css';

type QuestionnaireIntroProps = {
    content: {
        step2Header: string;
        step2Title: string;
        step2Instructions: string;
        step2Tip1: string;
        step2Tip2: string;
        step2Tip4: string;
        step2Tip5: string;
        step2Tip6: string;
        step2Header2: string;
        step2ProceedButton1: string;
    };
    setIntroPage: (value: boolean) => void;
}
export function QuestionnaireIntro(props: QuestionnaireIntroProps) {
    const {
        content,
        setIntroPage,
    } = props;
    const onClick = () => {
        setIntroPage(false);
    };

    return (
        <div className="QuestionnaireIntro">
            <div>
                <h1>
                    {content.step2Header}: {content.step2Title}
                </h1>
                <h2>{content.step2Instructions}</h2>
                <ul>
                    <li>{content.step2Tip1}</li>
                    <li>{content.step2Tip2}</li>
                </ul>
            </div>
            <div>
                <h1 className="header2">{content.step2Header2}</h1>
                <ul>
                    <li>{content.step2Tip4}</li>
                    <li>{content.step2Tip5}</li>
                    <li>{content.step2Tip6}</li>
                </ul>
            </div>
            <div className="ButtonContainer">
                <Button label={content.step2ProceedButton1} onClick={onClick} />
            </div>
        </div>
    );
}
