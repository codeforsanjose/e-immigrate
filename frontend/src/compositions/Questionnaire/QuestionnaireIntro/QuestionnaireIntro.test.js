import React from 'react';
import QuestionnaireIntro from './QuestionnaireIntro';

import renderer from 'react-test-renderer';

const content = {
    step2Header: 'Step 2',
    step2Header2: 'About the Screening Questionnaire',
    step2Instructions: 'Please complete the screening questionnaire',
    step2ProceedButton1: 'Begin Questionnaire',
    step2Tip1: 'It takes about 10 minutes',
    step2Tip2: 'Answer 25-30 questions',
    step2Tip4: 'All your information will be kept confidential',
    step2Tip5: 'Please answer truthfully and as best as you can remember',
    step2Tip6: 'Please complete ALL the questions to the best of your ability',
    step2Title: 'Questionnaire',
};

it('questionnaire intro renders correctly', () => {
    const tree = renderer
        .create(<QuestionnaireIntro content={content} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
