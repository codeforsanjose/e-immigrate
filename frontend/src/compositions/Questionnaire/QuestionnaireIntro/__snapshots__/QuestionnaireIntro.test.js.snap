// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`questionnaire intro renders correctly 1`] = `
<div
  className="QuestionnaireIntro"
>
  <div>
    <h1>
      Step 2
      : 
      Questionnaire
    </h1>
    <h2>
      Please complete the screening questionnaire
    </h2>
    <ul>
      <li>
        It takes about 10 minutes
      </li>
      <li>
        Answer 25-30 questions
      </li>
    </ul>
  </div>
  <div>
    <h1>
      About the Screening Questionnaire
    </h1>
    <ul>
      <li>
        All your information will be kept confidential
      </li>
      <li>
        Please answer truthfully and as best as you can remember
      </li>
      <li>
        Please complete ALL the questions to the best of your ability
      </li>
    </ul>
  </div>
  <div
    className="ButtonContainer"
  >
    <button
      className="buttonComponent"
      onClick={[Function]}
    >
      Begin Questionnaire
    </button>
  </div>
</div>
`;
